rfcs/react-components/components/dev-docs.md

```diff
@@ -0,0 +1,66 @@
+
+## Current issues:
+
+- Mixture of v6, 7, 8 and 9 specific docs.
+- Docs seemed to have stopped being updated in the early days of v9 development.
+- Mixture of process, set up, and reference docs.
+- RFC’s seem to be a large part of the Fluent process now, but there’s no clear way for an outsider to observe or contribute to them.
+- Use of Github’s wiki isn't the best technical solution.
+
+## Goals of the new system:
+
+- Be more accessible to developers in their work flow.
+- Help developers onboard faster without distracting core team members.
+- Provide technical documentation and guides for common patterns and practices that do not make sense to live in examples or code samples.
+- Focus attention on areas under active development (web and react components)
+
+## Non-Goals
+
+- Provide an ongoing communications platform for project statuses, road maps and planning.
+- Provide documentation relevant to other disciplines outside engineering.
+- Provide documentation around consuming the library. These should live externally in the Storybook.
+- Provide documentation around filing bugs or how to engage with the team.
+
+## Proposed changes
+
+### Place all dev documents in markdown files in a `__docs__` folder at the root of the repo.
+
+This would completely replace the wiki. There are two main advantages to this method:
+
+- Developers won’t have to context switch to access the documents.
+- Searching capabilities in VSCode tend to by wildly superior to those in Github.
+
+### Establish a new folder and docs structure:
+
+`__docs__`
+
+- rfcs
+
+  - \_rfc-process.md // Explainer on what RFCs are, are not and how they work.
+  - \_rfc-template.md // Starting point.
+  - some-historical-rfc.md
+  - // All the historical RFCs.
+
+- react-v9
+  - contributing
+    - dev-env.md // setting up your development environment for the first time
+    - dev-workflow.md // general end to end guidance of contributing a fix/feature/component end to end.
+    - command-cheat-sheet.md // simple table of all the commands we use daily.
+    - common-snags.md // a place to accumulate docs around common errors that we can link to.
+    - coding-style.md // any coding style guidelines not covered by the linter
+- internal_patterns
+  - // docs that explain patterns used internally in our components
+  - // can also include best practices recommendations
+- testing
+  - // docs that explain current testing processes, minimum bars, and helpful tips and tricks to spin up.
+- browser-support-matrix.md
+- ownership.md
+- stable-packages.md // a doc describing our approach to breaking changes, what qualifies, and how we make them
+- preview-packages.md // a doc explaining what a preview package is and how to create them
+
+web-components
+// Chat with Chris Holt here
+
+react-v8-and-below
+// Copy paste the exact same content from the current wiki.
```

Comment:
Do we want to copy this over or just point to a deprecated link to the wiki? (If that's a thing GitHub supports). Mostly just curious if we have the time/resources to move all that content over.

Reply:
My intent was to just copy paste it over. 

I don't want to devote too many cycles to tidying it, but felt there might be some value in keeping it around.

But I do want to fully remove the wiki though as having it up will cause further confusion.

Reply:
I think doing it the other way around would be helpful, copy over and then just leave one wiki article specifying that all dev-related documentation has moved into this folder.

Reply:
FYI, we have created a docusaurus based documentation site for the charting library. See here - https://microsoft.github.io/fluentui-charting-contrib/
This is inline with the practice followed by other open source projects in Mocrosoft and outside.